TimeStepping.pyx

```diff
@@ -127,6 +142,44 @@ cdef class TimeStepping:
 
         return
 
+    cpdef update_pressure_second(self, Grid.Grid Gr, PrognosticVariables.PrognosticVariables PV, DiagnosticVariables.DiagnosticVariables DV):
+        cdef:
+            Py_ssize_t i
+            Py_ssize_t press_shift = DV.get_varshift(Gr, 'density_perturbation_pressure')
+
+            Py_ssize_t whor_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve')
+            Py_ssize_t whor_rk0_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK0')
+            Py_ssize_t whor_rk1_shift = DV.get_varshift(Gr, 'wBudget_removeHorAve_RK1')
+
+            Py_ssize_t wpress_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient')
+            Py_ssize_t wpress_rk0_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK0')
+            Py_ssize_t wpress_rk1_shift = DV.get_varshift(Gr, 'wBudget_PressureGradient_RK1')
+
+
+        with nogil:
+            if self.rk_step == 0:
+                if self.t % self.dt_max == 0.0:
```

Comment:
Here, _self.dt_max_ is **not** the total timestep of your RK time step. It is an externally defined parameter that just sets a higher bound for the timestep, irrespective of the one given by the CFL condition. I do not think it should be here.